python/torch_mlir_e2e_test/test_suite/basic.py

```diff
@@ -3529,3 +3529,21 @@ def forward(self, x):
 @register_test_case(module_factory=lambda: MoveDimIntNegativeIndexModule())
 def MoveDimIntNegativeIndexModule_basic(module, tu: TestUtils):
     module.forward(tu.rand(3, 4, 2))
+
+
+# ==============================================================================
+
+class OneHotModule(torch.nn.Module):
+
+    def __init__(self):
+        super().__init__()
+    
+    @export
+    @annotate_args([None, ([-1], torch.long, True)])
+    def forward(self, x):
+        return torch.nn.functional.one_hot(x, num_classes=5)
+
+
+@register_test_case(module_factory=lambda: OneHotModule())
+def OneHotModule_basic(module, tu: TestUtils):
+    module.forward(torch.arange(0, 5))
```

Comment:
nit: this can be a random tensor to make the test more thorough: for example `tu.randint(10, high=5)`.